docs/topic/infrastructure/cluster-design.md

```diff
@@ -1,43 +1,93 @@
-of current number of running users. Picking an apporpriate node size and
-count here has a big effect.
-
-### On GKE
-
-GKE makes sizing this nodepool difficult, since `kube-system` components can take up quite
-a bit of resources. Even though the kind of clusters we run will most likely
-not stress components like `kube-dns` that much, there's no option to provide
-them fewer resource requests. So this will be our primary limitation in
-many ways.
-
-Adding [Config Connector](https://cloud.google.com/config-connector/docs/overview)
-or enabling [Network Policy](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy)
-requires more resources as well.
-
-With poorly structured experimentation, the current recommendation is to run
-3 `n1-highmem-2` instances for a cluster without config connector or network policy,
-or a single `n1-highmem-4` instance for a cluster with either of those options
-turned on. This needs to be better investigated.
+cluster's *base cost* - the amount of money it costs each day, regardless of
+current number of running users. Picking an appropriate instance type for the
+core nodes has a big effect.
+
+### CPU and Memory
+
+Kubernetes system Pods on GKE, EKS, and AKS will require different amount of
+allocatable capacity, and it can differ if we enable various features such as
+network policy enforcement.
+
+Typically we run out of allocatable Memory faster than CPU, which makes us
+benefit from nodes with a high memory to CPU ratio, such as the `n2-highmem`
+nodes on GCP or `r5` nodes on AWS.
+
+### Allocatable pods per node
+
+Node's of different types can only schedule so many pods on them.
+
+- GKE instance types supports 110 pods per node and typically won't have issues
+  with this, unless perhaps in a shared k8s cluster with many JupyterHub
+  installations.
+- EKS instance types support varying amounts of pods per node, and the smallest
+  nodes with two CPU cores typically only allows for 27 pods per node.
+
+### System pods anti-affinity
+
+Pods may be required to run on separate nodes. This has caused multiple nodes be
+required on GKE clusters that has `calico-typha` pods from the network policy
+enforcement.
+
+The `calico-typha` pods vary in amount, from only one to a few, depending on the
+number of nodes in the k8s cluster thanks to a HorizontalPodAutoscaler (HPA).
+The HPA can be re-configured via a k8s ConfigMap.
+
+```yaml
+# Below is a snippet from the ConfigMap calico-typha-horizontal-autoscaler in a
+# GKE cluster with network policy enforcement enabled. It can be edited, and
+# changes to it will be respected across k8s upgrades it seems.
+data:
+  ladder: |-
+    {
+      "coresToReplicas": [],
+      "nodesToReplicas":
+      [
+        [1, 1],
+        [2, 2],
+        [100, 3],
+        [250, 4],
+        [500, 5],
+        [1000, 6],
+        [1500, 7],
+        [2000, 8]
+      ]
+    }
+```
+
+On GKE clusters with network policy enforcement, we look to edit the
```

Comment:
Can you clarify this - is this something that just happens through terraform, deployer, does it need manual intervention (if so document steps),

Reply:
I opened https://github.com/2i2c-org/infrastructure/issues/2490 to offload further clarifications into another PR, right now we don't have a procedure or agreement on what to do.

An explicit action point in the linked issue is to update this text once its done!